plenum/common/timer.py

```diff
@@ -64,14 +67,15 @@ def wrapped_callback():
             self._timer.schedule(self._interval, self._callback)
 
         self._timer = timer
-        self._interval = interval
+        self._interval = None
+        self.update_interval(interval)
         self._callback = wrapped_callback
         self._active = False
         if active:
             self.start()
 
     def start(self):
-        if self._active:
+        if self._active or not self._interval:
```

Comment:
```suggestion
        if self._active or self._interval is None:
```

Reply:
In this case we will not catches interval = 0